src/ico/decoder.rs

```diff
@@ -16,6 +16,85 @@ use crate::png::PngDecoder;
 // The first eight bytes of a PNG file always contain the following (decimal) values:
 const PNG_SIGNATURE: [u8; 8] = [137, 80, 78, 71, 13, 10, 26, 10];
 
+/// Errors that can occur during decoding and parsing an ICO image or one of its enclosed images.
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub enum IcoDecoderError {
```

Comment:
Please keep this type private for now. Exposing it publicly makes it harder to move this into a separate crate or change it. We can still decide whether to expose it (and in what form) in a follow-up.

Reply:
Done